compiler/rustc_hir/src/target.rs

```diff
@@ -130,6 +131,29 @@ impl Target {
         }
     }
 
+    pub fn from_def_kind(def_kind: DefKind) -> Target {
+        match def_kind {
+            DefKind::ExternCrate => Target::ExternCrate,
+            DefKind::Use => Target::Use,
+            DefKind::Static(..) => Target::Static,
+            DefKind::Const => Target::Const,
+            DefKind::Fn => Target::Fn,
+            DefKind::Macro(..) => Target::MacroDef,
+            DefKind::Mod => Target::Mod,
+            DefKind::ForeignMod => Target::ForeignMod,
+            DefKind::GlobalAsm => Target::GlobalAsm,
+            DefKind::TyAlias => Target::TyAlias,
+            DefKind::OpaqueTy => Target::OpaqueTy,
+            DefKind::Enum => Target::Enum,
+            DefKind::Struct => Target::Struct,
+            DefKind::Union => Target::Union,
+            DefKind::Trait => Target::Trait,
+            DefKind::TraitAlias => Target::TraitAlias,
+            DefKind::Impl => Target::Impl,
+            _ => panic!("impossible case reached"),
```

Comment:
Why is it impossible?

Reply:
I'm sorry, this is an oversight. If this is used with `DefKind`s derived from `ItemId`s then I think it is not possible to see the other variants of `DefKind` here but I am not sure about `DefKind`s from `ImplItemId`, `TraitItemId`, etc.

Reply:
I added a comment so that users are aware of this until we merge `Target` and `DefKind`.